aiohttp/web_urldispatcher.py

```diff
@@ -672,6 +672,54 @@ def __repr__(self):
             prefix=self._prefix, app=self._app)
 
 
+class SubDomainResource(AbstractResource):
+
+    def __init__(self, subdomain, app):
+        super().__init__()
+        self._subdomain = subdomain
+        self._app = app
+
+    def add_prefix(self, prefix):
+        super().add_prefix(prefix)
+        for resource in self._app.router.resources():
+            resource.add_prefix(prefix)
+
+    def url_for(self, *args, **kwargs):
+        raise RuntimeError(".url_for() is not supported "
+                           "by subdomain sub-application root")
+
+    def url(self, **kwargs):
+        """Construct url for route with additional params."""
+        raise RuntimeError(".url() is not supported "
+                           "by subdomain sub-application root")
+
+    def get_info(self):
+        return {'app': self._app,
+                'subdomain': self._subdomain}
+
+    @asyncio.coroutine
+    def resolve(self, request):
+        if not request.headers['Host'] == self._subdomain:
```

Comment:
Again, pending the decision to have domain in place of subdomain, I guess using Yarl to parse the Host header here is also safer.